dev/tools/lib/git.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+import 'package:process/process.dart';
+
+/// A wrapper around git process calls that can be mocked for unit testing.
+class Git {
+  Git(this.processManager) : assert(processManager != null);
+
+  final ProcessManager processManager;
+
+  String getOutput(
+    List<String> args,
+    String explanation, {
+    @required String workingDirectory,
+  }) {
+    final ProcessResult result = _run(args, workingDirectory);
+    if ((result.stderr as String).isEmpty && result.exitCode == 0)
```

Comment:
sometimes stderr is output without it implying a non-zero exit. Could be a diagnostic message

Reply:
good point, as I should know (this was legacy code fwiw :P )